src/common-verifier/src/main/java/ee/ria/xroad/common/conf/globalconfextension/GlobalConfExtensions.java

```diff
@@ -37,25 +37,22 @@
  */
 @Slf4j
 public final class GlobalConfExtensions {
-
-    // Instance per thread, same way as in GlobalGonf/GlobalConfImpl.
-    // This way thread safety handling is same as in GlobalConf.
-    private static final ThreadLocal<GlobalConfExtensions> INSTANCE = new ThreadLocal<GlobalConfExtensions>() {
-        @Override
-        protected GlobalConfExtensions initialValue() {
-            return new GlobalConfExtensions();
-        }
-    };
-
     private OcspNextUpdate ocspNextUpdate;
     private OcspFetchInterval ocspFetchInterval;
+    private static volatile GlobalConfExtensions instance = null;
 
     /**
      * @return instance
      */
     public static GlobalConfExtensions getInstance() {
-        GlobalConfExtensions configuration = INSTANCE.get();
-        return configuration;
+        if (instance == null) {
+            synchronized (GlobalConfExtensions.class) {
+                if (instance == null) {
+                    instance = new GlobalConfExtensions();
+                }
+            }
+        }
+        return instance;
```

Comment:
These changes are not sound since the GlobalConfExtensions instance is not thread-safe, and now it is potentially shared by several concurrent threads. Since this task has been open quite some time, I suggest reverting these changes and creating a separate task of fixing GlobalConfExtensions.